packages/node_modules/@ciscospark/widget-roster/src/container.js

```diff
@@ -0,0 +1,164 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import classNames from 'classnames';
+import {autobind} from 'core-decorators';
+
+import {fetchAvatarsForUsers} from '@ciscospark/redux-module-avatar';
+
+import LoadingScreen from '@ciscospark/react-component-loading-screen';
+import Button from '@ciscospark/react-component-button';
+
+import {
+  ICON_TYPE_EXTERNAL_USER_OUTLINE,
+  ICON_TYPE_EXIT,
+  ICON_TYPE_WAFFLE,
+  default as Icon
+} from '@ciscospark/react-component-icon';
+
+import {getRosterWidgetProps} from './selector';
+import messages from './messages';
+
+import styles from './styles.css';
+
+import ParticipantList from './components/participant-list';
+
+/**
+ * RosterWidget Container
+ * @extends Component
+ */
+export class RosterWidget extends Component {
+
+  shouldComponentUpdate(nextProps) {
+    return nextProps.conversation.get(`participants`) !== this.props.conversation.get(`participants`) ||
+      nextProps.avatar.get(`items`) !== this.props.avatar.get(`items`);
+  }
+
+  componentWillUpdate(nextProps) {
+    this.props.fetchAvatarsForUsers(nextProps.participants.map((user) => user.userId), this.props.sparkInstance);
```

Comment:
For massive rooms, this will 429 the people endpoint until batching gets merged into js-sdk.